### PR TITLE
Fix list collector placeholders not resolving when preview enabled

### PR DESCRIPTION
**Pull request by petechd**
### What is the context of this PR?
This adds list collector placeholders resolution when preview questions is set to true in a schema .In `context.py` `placeholder_preview_mode` is now an arg of the init method which defaults to False and in `preview_context.py` and `section_preview_context.py` it's set to True.

### How to review
Preview flag added to list collector schema and functional test that checks if items are rendered on summary page.

### Checklist
* [ ] New static content marked up for translation
* [ ] Newly defined schema content included in eq-translations repo

### CODE REVIEW

**Comment by petechd** [comment]
[Bug: List collector placeholders not resolving when preview questions used](https://trello.com/c/im9QWuJI/5756-bug-list-collector-placeholders-not-resolving-when-preview-questions-used)

**Comment by petechd** [comment]
When preview flag is set to `true` introduction block is required.

File: app/views/contexts/preview_context.py [hunk 1]
```diff
@@ -27,6 +27,7 @@ def __init__(
         if not schema.preview_enabled:
             raise PreviewNotEnabledException
 
+        placeholder_preview_mode = True
```
Comment (MebinAbraham) [inline app/views/contexts/preview_context.py:30]: Minor, no need for this var, just pass it inline?.

File: schemas/test/en/test_list_collector_list_summary.json [hunk 2]
```diff
@@ -111,6 +121,202 @@
                     "id": "group",
                     "title": "Questions",
                     "blocks": [
+                        {
+                            "id": "introduction",
+                            "type": "Introduction",
```
Comment (MebinAbraham) [inline schemas/test/en/test_list_collector_list_summary.json:126]: If we are adding one, it should be very basic, this has so much content that isn't needed.

File: app/views/contexts/section_preview_context.py [hunk 3]
```diff
@@ -20,6 +20,7 @@ def __init__(
         response_metadata: MutableMapping,
         section_id: str,
     ):
+        placeholder_preview_mode = True
```
Comment (MebinAbraham) [inline app/views/contexts/section_preview_context.py:23]: Same as above, just pass it inline?.

File: schemas/test/en/test_list_collector_list_summary.json (hunk 2 above)
Comment (petechd) [reply schemas/test/en/test_list_collector_list_summary.json]: It is very basic now, literally one sentence/link on the page!

File: schemas/test/en/test_list_collector_list_summary.json [hunk 4]
```diff
@@ -111,6 +121,10 @@
                     "id": "group",
                     "title": "Questions",
                     "blocks": [
+                        {
+                            "id": "introduction",
+                            "type": "Introduction"
+                        },
```
Comment (MebinAbraham) [inline schemas/test/en/test_list_collector_list_summary.json:118]: We don't need it?. Preview as a feature can still be accessed?. Non blocker

**Review by MebinAbraham — APPROVED** [review]

File: schemas/test/en/test_list_collector_list_summary.json (hunk 4 above)
Comment (berroar) [reply schemas/test/en/test_list_collector_list_summary.json:118]: Think it's needed to pass validation?

**Review by berroar — APPROVED** [review]

File: schemas/test/en/test_list_collector_list_summary.json (hunk 4 above)
Comment (petechd) [reply schemas/test/en/test_list_collector_list_summary.json:118]: https://github.com/ONSdigital/eq-questionnaire-runner/pull/1151#issuecomment-1617744410

**Pull request merged**